Makefile

```diff
@@ -20,11 +20,12 @@ usage:
 	@echo "          alldoc: Builds Tock documentation for all boards"
 	@echo "           audit: Audit Cargo dependencies for all kernel sources"
 	@echo "              ci: Run all continuous integration tests"
-	@echo " emulation-setup: Setup QEMU for the emulation tests"
-	@echo " emulation-check: Run the emulation tests for supported boards"
 	@echo "           clean: Clean all builds"
+	@echo "          clippy: Runs the clippy code linter on all kernel sources"
+	@echo " emulation-check: Run the emulation tests for supported boards"
+	@echo " emulation-setup: Setup QEMU for the emulation tests"
 	@echo "          format: Runs the rustfmt tool on all kernel sources"
-	@echo "       formatall: Runs all formatting tools"
+	@echo "       formatall: Runs all formatting tools (rustfmt and clippy)"
```

Comment:
I find this confusing. `rustfmt` will modify the code by formatting it. But `clippy` is only a linter, and will not format the code at all.

Also, I think we have too many of these rules, which is too complex to reason about: `fmt`, `format`, `formatall`, `ci-formatting`, `ci-format`.

To me, there are two categories:
- A rule that actually modifies the code, which runs `./tools/run_cargo_fmt.sh`, and should be named `format`. I don't think we need `fmt` nor `formatall` spellings. There is no `ci` equivalent because CI only checks the code.
- Rules that check the code. I suggest:
  - `check-format` (or `format-check`), which runs `CI=true ./tools/run_cargo_fmt.sh diff`
  - `lint` (or `lints`), which runs both `check-format` and `clippy`
  - `ci-lint`, with the CI banner.

I'm not sure what `ci-documentation` has to do with any of these.

Reply:
`ci-format` is the grouping of stuff run by the github actions job with the name `ci-format`. I didn't want to break down every single little ci target we have to run in a separate job in github actions, but wanted to have a top level target people can run for when the job with that name fails.

`ci-documentation` is included in that because I consider validating a table of contents as a formatting check.

replacing `formatall` with `check-format` as you have described makes sense to me.

I don't see any need to get rid of the `fmt` spelling as some people may be used to using it.